src/serializers/encoder/v7/encode_graph.c

```diff
@@ -104,11 +104,22 @@ static PayloadInfo *_RdbSaveKeySchema(RedisModuleIO *rdb, GraphContext *gc) {
 																	   remaining_entities);
 			payloads = array_append(payloads, current_state_payload_info);
 			remaining_entities -= current_state_payload_info.entities_count;
-			if(remaining_entities > 0) {
+			if(remaining_entities >= 0) {
 				offset = 0; // New state offset is 0.
 				current_state++; // Advance in the states.
 			}
 		}
+		// Check if this is the last key
+		if(GraphEncodeContext_GetProcessedKeyCount(gc->encoding_context) ==
+		   GraphEncodeContext_GetKeyCount(gc->encoding_context) - 1) {
+			// This is the last key, no more entities to encode. Encode "empty" states + schema state.
+			while(current_state < ENCODE_STATE_FINAL) {
+				PayloadInfo current_state_payload_info = _StatePayloadInfo(gc, current_state, offset,
+																		   remaining_entities);
+				payloads = array_append(payloads, current_state_payload_info);
+				current_state++; // Advance in the states.
+			}
+		}
```

Comment:
Consider incorporating this logic within the above while loop, by:

Setting a local var
```
bool last_key = (GraphEncodeContext_GetProcessedKeyCount(gc->encoding_context) == GraphEncodeContext_GetKeyCount(gc->encoding_context) - 1);
```

Within the loop body set `remaining_entities` to UNLIMITED if last_key is true.